pkg/snapshot/snapshotter/snapshotter.go

```diff
@@ -740,3 +750,53 @@ func (ssr *Snapshotter) checkSnapstoreSecretUpdate() bool {
 	snapstoreHash[ssr.snapstoreConfig.Provider] = newSnapstoreSecretHash
 	return true
 }
+
+// IsTakingFullSnapRequiresAtStartup checks whether to take a full-snapshot or not during startup of backup-restore.
+func (ssr *Snapshotter) IsTakingFullSnapRequiresAtStartup(timeWindow float64) bool {
+	if ssr.PrevFullSnapshot == nil || ssr.PrevFullSnapshot.IsFinal || time.Since(ssr.PrevFullSnapshot.CreatedOn).Hours() > timeWindow {
+		return true
+	}
+
+	if !ssr.WasScheduledFullSnapshotMissed(timeWindow) {
+		return false
+	}
+	return ssr.IsNextFullSnapCanCrossTimeWindow(timeWindow)
+}
+
+// WasScheduledFullSnapshotMissed determines whether the preceding full-snapshot was missed or not.
+func (ssr *Snapshotter) WasScheduledFullSnapshotMissed(timeWindow float64) bool {
+	now := time.Now()
+	nextSnapSchedule := ssr.schedule.Next(now)
+
+	if miscellaneous.GetPrevDayScheduledSnapTime(nextSnapSchedule) == ssr.PrevFullSnapshot.CreatedOn {
```

Comment:
Looks like we're still assuming that the previous full snapshot must have been taken exactly 1 day before the next scheduled full snapshot. Is there no way to get the actual previous cron schedule? We can take inspiration from libraries like [supercronic](https://github.com/aptible/supercronic/tree/master/cronexpr) which allows getting the `Next`  cron scheduled time, and probably enhance it to get the previous cron scheduled time.

To me it looks like that will take a while to implement, so I'm still ok with the current implementation which assumes that full snapshot is taken once every day. But please add a `TODO` comment and create an issue for `respecting all cron schedules for full snapshots`, so that we can revisit the topic later and work on it if this PR needs to go in soon for the upcoming release to fix g/g alerting issues.

@unmarshall I think you were mentioning something about enhancing the existing https://github.com/robfig/cron package to provide `Next` and `Previous` cron scheduled times right?

Reply:
I will open a issue for this.